eurito_daps/packages/utils/openaire_utils.py

```diff
@@ -0,0 +1,165 @@
+from bs4 import BeautifulSoup
+import re
+from eurito_daps.production.orms import openaire_orm
+from eurito_daps.packages.utils import globals
+
+
+def link_record_with_project(record, recordObj):
+    '''A utility function, which links records with related EC projects and stores this linkage in the association table in the database
+
+    Args:
+        record (dict): contains metadata of a record (e.g. record of type "software" would store fields such as "title", "pid", "creator", etc.)
+        recordObj (ORM Object): a record returned by a query from the database (could be software, dataset, publication or ECProject record)
+    '''
+    #extract project codes
+    projectcodes = record['projectcodes']
+    #for each project code, create projectcode object,
+    #add a relationship between this projectcode and softwareObj in association table
+    for projectcode in projectcodes:
+        #find project with this code and create an object
+        projectObj = find_project_in_db(projectcode.text)
+        #if there is a related project, create a relationship between current software and found project
+        if projectObj:
+            #print ("projectObj.id exists", projectObj.id)
+            projectObj.software.append(recordObj)
+
+    return recordObj
+
+def find_project_in_db(inprojectcode):
+    '''A utility function, which returns ORM-type record objects from the database with the specified EC project code
+
+    Args:
+        inprojectcode (int): EC 6-digit project code

```

Comment:
variable naming conventions in python use `lower_camel_case`, so `in_project_code` would be better!